services/storage/src/simcore_service_storage/models.py

```diff
@@ -168,9 +178,19 @@ def convert_from_lower_case(cls, v):
             return f"{v}".upper()
         return v
 
+    @root_validator()
+    @classmethod
+    def when_directory_force_link_type_and_file_size(cls, values):
```

Comment:
Let me try to answer:
1. from the tests, this behaves as expected, being a directory it forces the defaults.
2. `files_size=None` means that the file's size was not yet computed. `file_size=UNDEFINED_SIZE` means it is unknown and it will not be provided in the future.

Reply:
if `None` has that meaning, then i think it should be documented.

Reply:
1. that was not my point :-) let's check that offline better. thx